packages/dht/src/connection/ConnectorFacade.ts

```diff
@@ -106,26 +106,37 @@ export class DefaultConnectorFacade implements ConnectorFacade {
                 const connectivityResponse = await this.websocketConnector!.checkConnectivity(false)
                 localPeerDescriptor = this.config.createLocalPeerDescriptor(connectivityResponse)
                 this.localPeerDescriptor = localPeerDescriptor
+                this.websocketConnector!.setLocalPeerDescriptor(localPeerDescriptor)
                 if (localPeerDescriptor.websocket === undefined) {
-                    logger.warn('ConnectivityCheck failed after autocertification, websocket server connectivity disabled')
+                    logger.warn('ConnectivityCheck failed after autocertification, disabling websocket server TLS')
+                    await this.restartWebsocketConnector({
+                        ...webSocketConnectorConfig,
+                        serverEnableTls: false
+                    })
                 }
-                this.websocketConnector!.setLocalPeerDescriptor(localPeerDescriptor)
             } catch (err) {
                 logger.warn('Failed to autocertify, disabling websocket server TLS')
-                await this.websocketConnector.destroy()
-                this.websocketConnector = new WebsocketConnector({
+                await this.restartWebsocketConnector({
                     ...webSocketConnectorConfig,
-                    serverEnableTls: false,
+                    serverEnableTls: false
                 })
-                await this.websocketConnector.start()
-                const connectivityResponse = await this.websocketConnector.checkConnectivity(false)
-                localPeerDescriptor = this.config.createLocalPeerDescriptor(connectivityResponse)
-                this.localPeerDescriptor = localPeerDescriptor
-                this.websocketConnector.setLocalPeerDescriptor(localPeerDescriptor)
             }
         }
         this.webrtcConnector.setLocalPeerDescriptor(localPeerDescriptor)
     }
+    
+    async restartWebsocketConnector(webSocketConnectorConfig: WebsocketConnectorConfig): Promise<void> {
+        await this.websocketConnector!.destroy()
+        this.websocketConnector = new WebsocketConnector({
+            ...webSocketConnectorConfig,
+            serverEnableTls: false,
+        })
+        await this.websocketConnector.start()
+        const connectivityResponse = await this.websocketConnector.checkConnectivity(false)
+        const localPeerDescriptor = this.config.createLocalPeerDescriptor(connectivityResponse)
+        this.localPeerDescriptor = localPeerDescriptor
+        this.websocketConnector.setLocalPeerDescriptor(localPeerDescriptor)
```

Comment:
We could have a helper method `setLocalPeerDescriptor` in `ConnectorFacade`, which would set all three fields. That way the values are always in sync, and the correct value of localPeerDescriptor will be updated to `webrtcConnector` too.

We'd use the method here an at line ~108. But wouldn't call it at line 125, i.e the `webrtcConnector` would receive the peerDescriptor a bit later (at the same time when websocket connector receives it). 

(Alternatively, if it is important that `webrtcConnector` can start asap, already before the `websocketConnector` is ready, we should update the correct `peerDescritor` also to that component.)

```

private setLocalPeerDescriptor(peerDescriptor: PeerDescriptor) {
    this.localPeerDescriptor = peerDescriptor
    this.websocketConnector!.setLocalPeerDescriptor(peerDescriptor)
    this.webrtcConnector!.setLocalPeerDescriptor(peerDescriptor)
}
```